infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InLogEntrySyncState.java

```diff
@@ -55,23 +55,25 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
                         "snapshot sync request";
                 cancelLogEntrySync(cancelCause);
                 LogReplicationState snapshotSyncState = fsm.getStates().get(LogReplicationStateType.IN_SNAPSHOT_SYNC);
-                snapshotSyncState.setTransitionEventId(event.getEventId());
+                // set the ID of the new snapshot sync to the incoming request ID
+                snapshotSyncState.setTransitionEventId(event.getMetadata().getRequestId());
                 ((InSnapshotSyncState)snapshotSyncState).setForcedSnapshotSync(event.getMetadata().isForcedSnapshotSync());
                 return snapshotSyncState;
             case SYNC_CANCEL:
                 // If cancel was intended for current log entry sync task, cancel and transition to new state
                 // In the case of log entry sync, cancel is caused by an encountered trimmed exception or any
                 // other exception while reading/sending.
-                if (transitionEventId == event.getMetadata().getRequestId()) {
+                if (fsm.isValidTransition(transitionEventId, event.getMetadata().getRequestId())) {
```

Comment:
The semantics here is different right? this check is for log entry sync, whose event id and eventMetadata id are also different